docs/website/contents/for-developers/Ticking.md

```diff
@@ -0,0 +1,98 @@
+# Ticking
+
+"Tick" is a common term in discrete systems for time passing.
+Within Ouroboros, time is divide into slots, and to _tick_ a state X means to advance X from one slot to a later slot.
+In the Cardano system, we tick either ledger states, protocol states, or both at once (see `ExtLedgerState`).
+
+The fundamental use case for ticking is to advance the ledger state between blocks.
+In Ouroboros, slots are often empty.
+The ledger state that resulted from some block thus needs to be ticked across some number of (empty!) slots before the next block can be applied to it.
+Some crucial things can happen when ticking between blocks: deadlines come and go, epochs end and begin, protocol parameters change, or even the ledger rules themselves might change.
+
+However, applying blocks is not the only reason the Consensus Layer ticks.
+If it were, then ticking wouldn't need to be separately identified: it'd just be the first of however many "hidden" steps are involved in applying a block.
+Instead, ticking is used in a few places, some of which don't involve immediately subsequent header/block application.
+
+- Ticking advances protocol state between headers.
+- Ticking advances ledger states between blocks (via `ExtLedgerState`, this also ticks the protocol state).
+- The leadership check ticks the protocol state from the selected chain's tip to the wall clock's current slot when checking whether this stake pool should mint a block.
+- The mempool ticks the selected chain's ledger state ahead by exactly one slot, since the hypothetical next block to extend the selection has to be strictly younger than the selection.
+  (Only Byron's Epoch Boundary Blocks can share their predecessor's slot, but EBBs also don't contain transactions.)
+- (Today's mint also ticks the ledger state up to the current slot, but that's actually unnecessary as of a recent change to the mempool-mint interface.)
```

Comment:
I'm ignorant of this change, what was it?

Reply:
I was thinking of https://github.com/IntersectMBO/ouroboros-consensus/pull/1175 `forgeShelleyBlock` in ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Forge.hs

The forge functions used to calculate which prefix could fit in a block. Now the Mempool handles that: it gives the forge functions the longest prefix that will fit.

.........

but that's not actually a big enough change to justify the text I've drafted here. I'll fix it.

Thanks for asking.